### PR TITLE
feat: gains `version.md` guidance

### LINKED ISSUE #18: Define and document our "release" process
I have a general question to @AlexAxthelm  and @cjyetman: what is our "release" process? 
* Initially, we would only bump `r2dii.*` version numbers once they were accepted to CRAN
* We then entered a situation where we "released" the `pacta.*` packages when they were used to build a "release candidate" for Transition Monitor. This was more or less just to facilitate "easily" re-building the image based on the same package versions, to be able to simulate, e.g. re-running a COP portfolio. 

As far as I can tell, this is now serving neither of the above, but instead the `WebApp` circumstance, so I guess this perhaps is a call to align on our definitions of a "release", otherwise I fear it ends up being a meaningless term. 

NIT: I will extract this into an issue in the `practices` repo.

_Originally posted by @jdhoffa in https://github.com/RMI-PACTA/workflow.portfolio.parsing/pull/17#discussion_r1478434437_

### PR DESCRIPTION
**Pull request by jdhoffa**
This PR documents much of our discussion around "Software Versioning" guidance.

Closes #18

### CODE REVIEW

File: version.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Semantic versioning (SemVer) is a versioning scheme that provides a simple set of rules and requirements that dictate how version numbers are assigned and incremented. The version number is a three-part number in the format of `MAJOR.MINOR.PATCH`.
+- `MAJOR` version when you make incompatible API changes,
+- `MINOR` version when you add functionality in a backwards-compatible manner, and
+- `PATCH` version when you make backwards-compatible bug fixes, or maintenance/ upkeep and other developer-facing improvements.
+
+
+## Calender Versioning
+
+Calender versioning (CalVer) is a versioning scheme that uses some portion of the date of release as the version number. There are different styles of CalVer. One is very similar to SemVer, however the `MAJOR` version is instead replaced with the year of release, e.g. `YYYY.MINOR.PATH`. Other variants can be found here: [CalVer](https://calver.org/).
```
Comment (cjyetman) [inline version.md:15]:
```suggestion
Calender versioning (CalVer) is a versioning scheme that uses some portion of the date of release as the version number. There are different styles of CalVer. One is very similar to SemVer, however the `MAJOR` version is instead replaced with the year of release, e.g. `YYYY.MINOR.PATCH`. Other variants can be found here: [CalVer](https://calver.org/).
```

File: version.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## R Packages
+
+R packages should be versioned using a Semantic Versioning (SemVer) compliant variant as described in the [R package versioning guide](https://r-pkgs.org/release.html#release-version).
```
Comment (cjyetman) [inline version.md:19]: maybe there should be a mention of the `0.1.4.9000` style of versioning that is widely used currently in our R packages?

Comment (jdhoffa) [reply version.md:19]: good idea!

Comment (jdhoffa) [reply version.md:20]: Resolved by adding a comment about that convention.

**Review by cjyetman — APPROVED** [review]

File: version.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Software Versioning
+
+Versioning is a critical aspect of software development. It is the process of assigning identifiers to a package to distinguish it from other versions of the same package. In contrast to a "git hash", which serves as an immutable machine-friendly identifier of the state of the software, the version number serves more to communicate the function of the package to (human) users. This document outlines our practices for versioning a package.
```
Comment (AlexAxthelm) [inline version.md:3]:
```suggestion
Versioning is a critical aspect of software development. It is the process of assigning identifiers to a package to distinguish it from other versions of the same package. In contrast to a "git hash", which serves as an immutable machine-friendly identifier of the state of the code, the version number serves more to communicate the function of the system to (human) users. This document outlines our practices for versioning a package.
```

File: version.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+## Semantic Versioning
+
+Semantic versioning (SemVer) is a versioning scheme that provides a simple set of rules and requirements that dictate how version numbers are assigned and incremented. The version number is a three-part number in the format of `MAJOR.MINOR.PATCH`.
```
Comment (AlexAxthelm) [inline version.md:7]:
```suggestion
Semantic versioning ([SemVer](https://semver.org/)) is a versioning scheme that provides a simple set of rules and requirements that dictate how version numbers are assigned and incremented. The version number is a three-part number in the format of `MAJOR.MINOR.PATCH`.
```

File: version.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+- `MAJOR` version when you make incompatible API changes,
```
Comment (AlexAxthelm) [inline version.md:8]:
```suggestion
- `MAJOR` version increments when you make backwards-incompatible changes
```

File: version.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+- `MINOR` version when you add functionality in a backwards-compatible manner, and
```
Comment (AlexAxthelm) [inline version.md:9]:
```suggestion
- `MINOR` version when you add functionality in a backwards-compatible manner
```

File: version.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+- `PATCH` version when you make backwards-compatible bug fixes, or maintenance/ upkeep and other developer-facing improvements.
```
Comment (AlexAxthelm) [inline version.md:10]:
```suggestion
- `PATCH` version increments when you make backwards-compatible bug fixes, or maintenance/ upkeep and other developer-facing improvements.
```

**Review by AlexAxthelm — DISMISSED** [review]
Minor suggestions, non-blocking

File: version.md (hunk 6 above)
Comment (jacobvjk) [reply version.md]: for consistency, you might want to suggest `MINOR version **increments**`  here

File: version.md (hunk 3 above)
Comment (jdhoffa) [reply version.md:3]: Happy to convert `software->code`, but can you explain why you prefer not to use "package"?
I use that term already a lot in the first sentence, so probably best to be self-consistent if possible?

**Review by jacobvjk — APPROVED** [review]

File: version.md (hunk 3 above)
Comment (AlexAxthelm) [reply version.md:3]: Not all software is packages.

Maybe a better phrasing would be something along the lines of

> `the version number serves to communicate the behavior of the code to humans`

**Review by jacobvjk — APPROVED** [review]

**Review by MonikaFu — APPROVED** [review]
LGTM

**Pull request merged**